packages/utilities/src/utils/common.ts

```diff
@@ -199,3 +202,56 @@ export const createRuntimeVariables = (remotes: Remotes) => {
     return acc;
   }, {} as Record<string, string>);
 };
+
+/**
+   * Returns initialized webpack RemoteContainer.
+   * If its' script does not loaded - then load & init it firstly.
+   */
+export const getContainer = async (remoteContainer: string | RemoteData) : Promise<WebpackRemoteContainer> => {
+
+  if(!remoteContainer) {
+    throw Error(`Remote container options is empty`);
+  }
+  if(typeof remoteContainer === 'string') {
+    if (window[remoteContainer]) {
+      return window[remoteContainer];
+    }
+  } else {
+    if (window['uniqueKey' as keyof typeof  remoteContainer]) {
+      return window['uniqueKey' as keyof typeof  remoteContainer];
+    }
+
+    try {
+      const container = await injectScript({
+        global: remoteContainer.global,
+        url: remoteContainer.url,
+      });
+
+      if (container) {
+        return container;
+      }
+      throw Error(`Remote container ${remoteContainer.url} is empty`);
+    } catch (e) {
+       throw e;
+    }
+  }
+}
+
+  /**
+  * Return remote module from container.
+  * If you provide `exportName` it automatically return exact property value from module.
+  *
+  * @example
+  *   remote.getModule('./pages/index', 'default')
+  */
+export const getModule = async ({ remoteContainer, modulePath, exportName }: GetModuleOptions) => {
+  const container = await getContainer(remoteContainer);
+  const modFactory = await container.get(modulePath);
+  if (!modFactory) return undefined;
+  const mod = modFactory();
+  if (exportName) {
+    return mod && typeof mod === 'object' ? mod[exportName] : undefined;
```

Comment:
Can we use optional chaining and nullish coalescing?

Reply:
@ExProbitasFiducia in fact i only had isolate the source code and bring to the utilities, i haven't develop the code 😅 

@ScriptedAlchemy using  optional chaining and nullish coalescing here would have the same effect or you had avoided to use because you've found any problem?

Reply:
See if it compiles down to ES5 or not, if it keeps the modern syntax then id avoid using it since it will break some browsers / node versions

Reply:
It **should** according to this SO post, but it could be down to config; it depends what the target we're compiling to is. If we're targeting es5 in our config then it ought to work though; full disclosure I have not checked that nor have I looked at the config.